huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go

```diff
@@ -102,45 +407,489 @@ func resourceGaussDBProxyUpdate(ctx context.Context, d *schema.ResourceData, met
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		} else {
+			err = reduceGaussDBMySQLProxyNumber(ctx, d, client, oldNum.(int)-newNum.(int))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	if d.HasChange("proxy_name") {
+		err = updateGaussDBMySQLProxyName(d, client)
+		if err != nil {
+			return diag.FromErr(err)
 		}
+	}
 
-		enlargeSize := newnum.(int) - oldnum.(int)
-		enlargeProxyOpts := instances.EnlargeProxyOpts{
-			NodeNum: enlargeSize,
+	if d.HasChange("port") {
+		err = updateGaussDBMySQLPort(ctx, d, client)
+		if err != nil {
+			return diag.FromErr(err)
 		}
+	}
 
-		lp, err := instances.EnlargeProxy(client, d.Id(), enlargeProxyOpts).ExtractJobResponse()
+	if d.HasChanges("master_node_weight", "readonly_nodes_weight") {
+		err = updateGaussDBMySQLNodesWeight(ctx, d, client)
 		if err != nil {
-			return diag.Errorf("error enlarging gaussdb_mysql_proxy: %s", err)
+			return diag.FromErr(err)
 		}
+	}
 
-		if err = instances.WaitForJobSuccess(client, int(d.Timeout(schema.TimeoutUpdate)/time.Second), lp.JobID); err != nil {
-			return diag.Errorf("error waiting for gaussdb_mysql_proxy job: %s", err)
+	if d.HasChanges("new_node_auto_add_status", "new_node_weight") {
+		err = updateGaussDBMySQLNewNode(d, client)
+		if err != nil {
+			return diag.FromErr(err)
 		}
 	}
 
 	return resourceGaussDBProxyRead(ctx, d, meta)
 }
 
-func resourceGaussDBProxyDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func updateGaussDBMySQLFlavor(ctx context.Context, d *schema.ResourceData, client *golangsdk.ServiceClient) error {
+	var (
+		httpUrl = "v3/{project_id}/instances/{instance_id}/proxy/{proxy_id}/flavor"
+	)
+	updatePath := client.Endpoint + httpUrl
+	updatePath = strings.ReplaceAll(updatePath, "{project_id}", client.ProjectID)
+	updatePath = strings.ReplaceAll(updatePath, "{instance_id}", d.Get("instance_id").(string))
+	updatePath = strings.ReplaceAll(updatePath, "{proxy_id}", d.Id())
+
+	updateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	updateOpt.JSONBody = utils.RemoveNil(buildUpdateGaussDBMySQLFlavorBodyParams(d))
+
+	updateResp, err := client.Request("PUT", updatePath, &updateOpt)
+	if err != nil {
+		return fmt.Errorf("error updating GaussDB MySQL flavor: %s", err)
+	}
+
+	updateRespBody, err := utils.FlattenResponse(updateResp)
+	if err != nil {
+		return err
+	}
+
+	jobId := utils.PathSearch("job_id", updateRespBody, nil)
+	if jobId == nil {
+		return fmt.Errorf("error updating GaussDB MySQL flavor: job_id is not found in API response")
+	}
+
+	err = checkGaussDBMySQLProxyJobFinish(ctx, client, jobId.(string), d.Timeout(schema.TimeoutUpdate))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildUpdateGaussDBMySQLFlavorBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"flavor_ref": d.Get("flavor"),
+	}
+	return bodyParams
+}
```

Comment:
It's no need to use d as function param when only use one of the values. Please check the same problems in those functions: buildEnlargeGaussDBMySQLProxyNumberBodyParams, buildUpdateGaussDBMySQLProxyNameBodyParams, buildUpdateGaussDBMySQLPortBodyParams, buildDeleteGaussDBMySQLProxyBodyParams

```suggestion
func buildUpdateGaussDBMySQLFlavorBodyParams(flavor string) map[string]interface{} {
	bodyParams := map[string]interface{}{
		"flavor_ref": flavor,
	}
	return bodyParams
}
```